internal/provider/resource_gitlab_personal_access_tokens.go

```diff
@@ -0,0 +1,265 @@
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Printf("[DEBUG] created gitlab PersonalAccessToken %d - %s for user ID %d", personalAccessToken.ID, *options.Name, userId)
+
+	tokenId := strconv.Itoa(personalAccessToken.ID)
+	userIdString := strconv.Itoa(userId)
+
+	d.SetId(buildTwoPartID(&userIdString, &tokenId))
+	// NOTE: the token can only be read once after creating it
+	d.Set("token", personalAccessToken.Token)
+
+	return resourceGitlabPersonalAccessTokenRead(ctx, d, meta)
+}
+
+func resourceGitlabPersonalAccessTokenRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	userId, tokenId, err := parseTwoPartID(d.Id())
+	if err != nil {
+		return diag.Errorf("Error parsing ID: %s", d.Id())
+	}
+
+	userIdInt, err := strconv.Atoi(userId)
+	if err != nil {
+		return diag.Errorf("Error converting user ID to string: %v", userId)
+	}
+
+	client := meta.(*gitlab.Client)
+
+	personalAccessTokenId, err := strconv.Atoi(tokenId)
+	if err != nil {
+		return diag.Errorf("%s cannot be converted to int", tokenId)
+	}
+
+	log.Printf("[DEBUG] read gitlab PersonalAccessToken %d, user ID %s", personalAccessTokenId, userId)
+
+	personalAccessToken, err := resourceGitlabPersonalAccessTokenFind(ctx, client, userIdInt, personalAccessTokenId)
+	if errors.Is(err, errResourceGitlabPersonalAccessTokenNotFound) {
+		log.Printf("[DEBUG] failed to read gitlab PersonalAccessToken %d, user ID %s", personalAccessTokenId, userId)
+		d.SetId("")
+	}
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.Set("user_id", userIdInt)
+	d.Set("name", personalAccessToken.Name)
+	if personalAccessToken.ExpiresAt != nil {
+		d.Set("expires_at", personalAccessToken.ExpiresAt.String())
+	}
+	d.Set("active", personalAccessToken.Active)
+	d.Set("created_at", personalAccessToken.CreatedAt.Format(time.RFC3339))
+	d.Set("revoked", personalAccessToken.Revoked)
+
+	err = d.Set("scopes", personalAccessToken.Scopes)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func resourceGitlabPersonalAccessTokenDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	userId, tokenId, err := parseTwoPartID(d.Id())
+	if err != nil {
+		return diag.Errorf("Error parsing ID: %s", d.Id())
+	}
+
+	userIdInt, err := strconv.Atoi(userId)
+	if err != nil {
+		return diag.Errorf("Error converting user ID to string: %v", userId)
+	}
+
+	client := meta.(*gitlab.Client)
+
+	personalAccessTokenId, err := strconv.Atoi(tokenId)
+	if err != nil {
+		return diag.Errorf("%s cannot be converted to int", tokenId)
+	}
+
+	log.Printf("[DEBUG] Delete gitlab PersonalAccessToken %s", d.Id())
+	_, err = client.PersonalAccessTokens.RevokePersonalAccessToken(personalAccessTokenId, gitlab.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Printf("[DEBUG] Waiting for PersonalAccessToken %s to finish deleting", d.Id())
+
+	err = resource.RetryContext(ctx, 5*time.Minute, func() *resource.RetryError {
```

Comment:
Did you try it without this retry first? As long as the tests pass, we do not need it.

Reply:
I thought there might be an underlying issue with responses in GitLab itself for tokens.  I can remove this and test it out, are we sure that that this isn't needed on a real GitLab instance?

Reply:
During my GitPod acceptance testing it appears that the retry is not needed

Reply:
I think we should remove it. I had to add it in the project_access_token resource because the `CheckDestroy` test failed consistently without it.

I added a comment to this GitLab issue here so we can get more info: https://gitlab.com/gitlab-org/gitlab/-/issues/352299#note_898951315